src/components/Facility/FacilityHome.tsx

```diff
@@ -105,23 +106,24 @@ export const FacilityHome = ({ facilityId }: Props) => {
       pathParams: { id: facilityId },
     }),
   });
+  const { mutate: deleteFacility } = useMutation({
+    mutationFn: () =>
+      mutate(routes.deleteFacility, { pathParams: { id: facilityId } })(null),
+    onSuccess: () => {
+      toast.success(t("deleted_successfully", { name: facilityData?.name }));
+      navigate("/facility");
+    },
+    onError: () => {
+      toast.error(t("delete_failed", { name: facilityData?.name }));
+    },
+  });
 
-  const handleDeleteClose = () => {
-    setOpenDeleteDialog(false);
+  const handleDeleteSubmit = () => {
+    deleteFacility();
   };
 
-  const handleDeleteSubmit = async () => {
-    await request(routes.deleteFacility, {
-      pathParams: { id: facilityId },
-      onResponse: ({ res }) => {
-        if (res?.ok) {
-          toast.success(
-            t("deleted_successfully", { name: facilityData?.name }),
-          );
-        }
-        navigate("/facility");
-      },
-    });
+  const handleDeleteClose = () => {
+    setOpenDeleteDialog(false);
   };
```

Comment:
why not just call `setOpenDialog` directly instead of creating a function and calling that function?